### DISCUSSION TITLE
Automated reporting in a Quarto blog

### DISCUSSION THREAD

**Discussion by VirginiaWoloshen** [quarto-dev/quarto-cli#2261] [Q&A]
Currently, I've been using a parameterized quarto template to generate reports. I pass the parameter to 

quarto_render(input = "template.qmd", execute_params = list(param1 = param_example), output_file = paste("./", param_example, "_report.html", sep = ""))

but this renders the html file which is fine for one off events but I would like to auto-generate a blog post on a Quarto blog with this. Can the output_file be an .qmd file (required for blog post listings) or is there some way for me to do this dynamically?

**Comment by mcanouil** [comment]
I am not sure to follow, you want to create a Quarto document from a Quarto document?
Could you explain a bit more your use case?

**Reply by VirginiaWoloshen** [reply]
mainly it would be for automated reporting. We have weekly data that is essentially reported in the same format so it would be nice to automatically generate these. Having it in a blog format would be nice so the team can view, search old reports and subscribe. I think I found a way using writeLines which works for now. I have written a qmd document as a template that is parameterized so all I need to do is change that one parameter (currently using str_replace) then writeLines to posts/

**Reply by mcanouil** [reply]
Then no rendering needed at all, except the blog part.
I'll make a reproducible demo later.

**Reply by mcanouil** [reply]
Here a reproducible example (this is not the only way):

1. Create a website
   ```bash
   quarto create-project --type website:blog
   ```

2. Create a template file `_template-report.qmd`
   ````
   ---
   categories: [reports]
   ---
 
   This is a post with executable code.
 
   ```{r}
   my_data <- get(data(list = params[["data"]]))
   knitr::kable(head(my_data))
   ```
   ````

3. Create an R function to fills the YAML, e.g., `title`, `author`, `date`, and `params`
   ```r
   create_report_post <- function(title, author, date, data, blog_dir = "posts") {
     post_dir <- gsub("-+", "-", gsub("[[:punct:]⁠ ]+", "-", tolower(title)))
     post_dir <- gsub("^-|-$", "", post_dir)
     dir.create(file.path(blog_dir, post_dir))
     writeLines(
       text = c(
         "---",
         yaml::as.yaml(
           list(
             title = title,
             author = author,
             date = date,
             params = list(
               data = data
             )
           )
         ),
         readLines("_template-report.qmd")[-1]
       ),
       con = file.path(blog_dir, post_dir, "index.qmd")
     )
   }
   ```
4. Generate a new post and render the website:
   ```r
   create_report_post(
     title = "April Report",
     author = "Firstname Lastname",
     date = format(Sys.Date(), "%Y-%m-%d"),
     data = "mtcars"
   )
   quarto::quarto_render(as_job = FALSE)
   ````
